src/components/Button/LinkButton.tsx

```diff
@@ -0,0 +1,63 @@
+import Link from 'next/link';
+import { ReactNode } from 'react';
+
+interface ButtonLinkProps {
+  href: string;
+  className?: string;
+  children?: ReactNode;
+  disabled?: boolean;
+  onClick?: () => void;
+  onSubmit?: () => void;
+  type?: 'button' | 'submit';
+}
+
+function getClassName(
+  className: string | undefined,
+  disabled: boolean
+): string {
+  const classMap: Record<string, string> = {
+    button_large: disabled ? 'button_large_active' : 'button_large_disActive',
+    button_medium: disabled
+      ? 'button_medium_active'
+      : 'button_medium_disActive',
+    button_small: disabled ? 'button_small_active' : 'button_small_disActive',
+    button_large_fill: disabled
+      ? 'button_large_active flex-1'
+      : 'button_large_disActive flex-1',
+  };
+
+  return classMap[className ?? ''] ?? className ?? '';
+}
+
+function LinkButton({
+  children,
+  className,
+  href,
+  onClick,
+  onSubmit,
+  disabled = false,
+  type = 'button',
+}: ButtonLinkProps) {
+  const finalClassName = getClassName(className, disabled);
+
+  const buttonProps = {
+    className: finalClassName,
+    onClick,
+    onSubmit,
+    type,
+  };
+
+  return (
+    <>
+      {disabled ? (
+        <Link href={href}>
+          <button {...buttonProps}>{children}</button>
+        </Link>
+      ) : (
+        <button {...buttonProps}>{children}</button>
+      )}
+    </>
+  );
```

Comment:
저였어도 이렇게 했을거같은데 어제 멘토님이 말씀해주신 오류잡는부분에서
isLoading, error 같은상태를 props로 받아서 그상태일때 리턴하는 부분을 따로 지정하는걸 봤을때 저렇게도 할수있구나 싶었거든요

```suggestion
if (disabled) {
    return (
        <Link href={href}>
          <button {...buttonProps}>{children}</button>
        </Link>
    )
}

return (
    <button {...buttonProps}>{children}</button>
)
```

이방법도 되게 괜찮은 방법같지않나요? 한번 써보고싶었어요

Reply:
오 저번 멘토링때 들은 내용 되게 좋은거 같아요!! 수정해볼게요!!